packages/grid/x-data-grid/src/internals/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -243,7 +243,12 @@ function GridFilterForm(props: GridFilterFormProps) {
 
   const handleDeleteFilter = () => {
     if (rootProps.disableMultipleColumnsFiltering) {
-      applyFilterChanges({ ...item, value: undefined });
+      if (item.value === undefined) {
```

Comment:
Note that it only deletes the last item when clicking twice. In a string column, the empty input means `item.value=""` which falls into the else branch.

![msedge_dvGxFA5EbL](https://user-images.githubusercontent.com/42154031/155005246-08974267-7b0f-4942-abf2-0e47ce761574.gif)

Reply:
And I assume the isAnyOf after being cleaned is an empty array, so we have two more cases to check, but we can not use `!item.value` due to boolean filters